dd-sdk-android/src/test/kotlin/com/datadog/android/rum/internal/domain/scope/RumActionScopeTest.kt

```diff
@@ -259,21 +270,20 @@ internal class RumActionScopeTest {
         fakeEvent = RumRawEvent.StartResource(key, url, method, emptyMap())
         val result = testedScope.handleEvent(fakeEvent, mockWriter)
         Thread.sleep(TEST_INACTIVITY_MS * 2)
-        fakeEvent = RumRawEvent.StopResourceWithError(
+        val fakeEvent2 = RumRawEvent.StopResourceWithError(
             key2,
             statusCode,
             message,
             source,
             throwable,
             emptyMap()
         )
-        val result2 = testedScope.handleEvent(fakeEvent, mockWriter)
+        val result2 = testedScope.handleEvent(fakeEvent2, mockWriter)
         Thread.sleep(TEST_INACTIVITY_MS * 2)
         val result3 = testedScope.handleEvent(mockEvent(), mockWriter)
 
         // Then
-        verifyZeroInteractions(mockParentScope, mockWriter)
```

Comment:
we now have non-zero interactions with `mockParentScope`?

Reply:
This test was flaky, I discovered it during my tests in this PR. Yes because we introduced the concept of `Continuous, NonContinuous` ActionScope. In case the scope is not continuous (not waiting for stop) it will still try to send the event in this case but it will not be able to due to lack of `sideEffects` and therefore it will go into the `else` branch calling :
```
  parentScope.handleEvent(
                RumRawEvent.ActionDropped(getRumContext().viewId.orEmpty()),
                writer
            )
```